src/server/pfs/server/driver.go

```diff
@@ -1641,6 +1641,11 @@ func (d *driver) fillNewBranches(txnCtx *txncontext.TransactionContext, branch *
 // This algorithm updates every branch in branchInfo's Subvenance, new Provenance, and old Provenance.
 // Complexity is O(m*n*log(n)) where m is the complete Provenance of branchInfo, and n is the Subvenance of branchInfo
 func (d *driver) computeBranchProvenance(txnCtx *txncontext.TransactionContext, branchInfo *pfs.BranchInfo, oldDirectProvenance []*pfs.Branch) error {
+	for _, p := range branchInfo.DirectProvenance {
```

Comment:
I think the check just needs to be if a user attempts to create a branch with itself in the provenance. The multi-branch cycles should be handled by the DAG validation.